web/src/people.ts

```diff
@@ -311,9 +338,7 @@ export function user_ids_string_to_emails_string(user_ids_string: string): strin
 
     emails = emails.map((email) => email.toLowerCase());
 
-    emails.sort();
-
-    return emails.join(",");
+    return sort_emails(emails, user_ids).join(",");
```

Comment:
The `sort_emails` function is used here and in `pm_reply_to`. Both these functions are mainly used to set the url hash and to set the `private_message_recipient` field which is eventually used to set the pills in recipient box and probably the compose box placeholder as well.

For urls, we eventually sort the user ids at last, so the order of emails should not matter. So, I think we can probably just add code to make sure the pills are rendered in sorted order and we wouldn't need to sort emails here.

Reply:
There are multiple functions for adding pills ( e.g., appendValidateData, appendPill, InsertManyPills ) but not any function  where we can access all pills together to check the order of them. We can make one function to check and run it every time data is refreshed or page is reloaded. But I find it better to have a data already sorted accordingly through this function.

Reply:
We could probably sort the emails in `compose_state.private_message_recipient` before calling `compose_pm_pill.set_from_emails`.

Though one thing that feels unnecessary here is that we first get the user object from IDs to get email and then we again get user objects from emails to sort them. We could probably just sort it initially when fetching user object from ID and be fine with a bit of duplication.